### ISSUE TITLE
Excel export of Excercises - Results crashes after upgrading to PHP 8.1

### ISSUE THREAD

**Issue by IkkeMus** [chamilo/chamilo-lms#4541]
When exporting as excel-format (csv seems to work, but is incompatible with national letters) I get the error message:

Fatal error: Array and string offset access syntax with curly braces is no longer supported in /home/5/m/.../www/chamilo-1.11.16/vendor/phpoffice/phpexcel/Classes/PHPExcel/Shared/String.php on line 526

It worked flawlessly with PHP 7.3. I guess this is merely a compatibility issue with PHP 8.x which could easily be resolved?

**Comment by ywarnier** [comment]
Hi @IkkeMus 
We don't officially support 8.0 (and even less 8.1) with the 1.11.x branch. We've made limited testing with 8.0 and it works (provided you don't run a composer update), and it will work better with the soon-to-be-released 1.11.18, but 8.1 is a bit too far for this branch. This is due to the multitude of libraries we depend on and that are not PHP8-ready. Chamilo itself works quite fine with 8.0.
We are running to release a beta of Chamilo 2.0 in the first half of this year. This one will only support 8.0 and above, and will support them fully, but there's still a lot of work to be done.

**Comment by IkkeMus** [comment]
cant wait to get my hands on 1.11.18! Keep up the good work. How can I sign up for being a translator (Norwegian)?

**Comment by ywarnier** [comment]
@IkkeMus You can sign up on https://translate.chamilo.org and ask to translate from English to Norwegian, then ping me back here to authorize you. Please note the translation system will change when we reach beta stage with 2.0.

**Comment by IkkeMus** [comment]
Application sent..

**Comment by IkkeMus** [comment]
BTW Chamilo 1.11.18 will also not export Exercise results in xls-format on PHP 8.1, csv works.

Still testing, but seems like 1.11.18 works like a charm on PHP 8.1. Awesome! Woohoo! Thank you!

**Comment by ywarnier** [comment]
> BTW Chamilo 1.11.18 will also not export Exercise results in xls-format on PHP 8.1, csv works.

Yes, there seems to be something wrong with the library to do the XLS export. Sorry. Will have to wait for 2.0.

> Still testing, but seems like 1.11.18 works like a charm on PHP 8.1. Awesome! Woohoo! Thank you!

Good to know. Thanks!

**Comment by ywarnier** [comment]
@IkkeMus btw, translations are now managed (for the master branch - i.e. v2.0) through PRs on the translations/messages.*.po files.

Regarding XLS exports, v2.0 should work like a charm now. We've also updated the XLS library in v1.11.38 and it both works better and is more compatible with later version of PHP (tested with PHP 8.3).
